sentry/src/main/java/io/sentry/TransactionOptions.java

```diff
@@ -121,6 +133,26 @@ public void setWaitForChildren(boolean waitForChildren) {
     return idleTimeout;
   }
 
+  /**
+   * Sets the deadlineTimeout. If set, an transaction and it's child spans will be force-finished
+   * with status {@link SpanStatus#DEADLINE_EXCEEDED} in case the transaction isn't finished in
+   * time.
+   *
+   * @param deadlineTimeoutMs - the deadlineTimeout, in ms - or null if no deadline should be set
+   */
+  public void setDeadlineTimeout(@Nullable Long deadlineTimeoutMs) {
```

Comment:
Not sure if we should make this public

Reply:
Do we have to align with other SDKs on this?
Otherwise i think it's fine to make it public

Reply:
I'd keep it internal for now, it's always easier to make it public later if needed.